components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/ManagementApiParser.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ */
+package org.wso2.micro.integrator.management.apis;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.impl.builder.StAXOMBuilder;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.micro.core.util.CarbonException;
+import org.wso2.micro.integrator.core.util.MicroIntegratorBaseUtils;
+import org.wso2.securevault.SecretResolverFactory;
+
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLStreamException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ * This class reads through internal-apis.xml and parses the values.
+ */
+public class ManagementApiParser {
+
+    private static final Log LOG = LogFactory.getLog(ManagementApiParser.class);
+    private static OMElement managementApiElement;
+
+    public ManagementApiParser() {
+    }
+
+    /**
+     * Method to get the File object representation of the internal-apis.xml file.
+     *
+     * @return File object representation
+     */
+    public static File getConfigurationFile() {
+        return new File(MicroIntegratorBaseUtils.getCarbonConfigDirPath(), "internal-apis.xml");
+    }
+
+    /**
+     * Method to get absolute path of the internal-apis.xml file.
+     *
+     * @return The absolute path for the file
+     */
+    public static String getConfigurationFilePath() {
+        return new File(MicroIntegratorBaseUtils.getCarbonConfigDirPath(), "internal-apis.xml").getAbsolutePath();
+    }
+
+    public static OMElement getManagementApiElement() throws IOException, CarbonException, XMLStreamException,
+            ManagementApiUndefinedException {
+        if (Objects.nonNull(managementApiElement)) {
+            return managementApiElement;
+        }
+        LOG.debug("Parsing " + getConfigurationFilePath() + " for the first time.");
```

Comment:
We need to check whether debug is enabled here .

Reply:
done